src/scheduler/resv_info.c

```diff
@@ -2035,8 +2043,8 @@ check_vnodes_unavailable(resource_resv *resv)
 			remove_ptr_from_array(resv->orig_nspec_arr, chunks_to_remove[i]);
 		}
 	}
-
-	qsort(resv->orig_nspec_arr, count_array(resv->orig_nspec_arr), sizeof(nspec *), cmp_nspec);
+	/* move all nspec with down nodes at the back of orig_ncpec_arr */
```

Comment:
While this is true, it doesn't explain why you are doing it.  Maybe something more like: "Move all the specific chunks ahead of the generic chunks.  This will ensure that we get all the nodes back we need to, before we start looking for replacements"